logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/UserSessionScope.kt

```diff
@@ -287,6 +289,13 @@ abstract class UserSessionScopeCommon(
             authenticatedDataSourceSet.authenticatedNetworkContainer.notificationApi, eventInfoStorage, clientRepository
         )
 
+    internal val keyPackageManager: KeyPackageManager
+        get() = KeyPackageManagerImpl(
```

Comment:
By keeping it in a `get() =` it will never be built unless `keyPackageManager` is called.

Maybe just:

```kotlin
internal val keyPackageManager: KeyPackageManager = KeyPackageManagerImpl(
    syncRepository,
    lazy { keyPackageRepository },
    lazy { client.refillKeyPackages }
)
```

This way it will start observing `syncRepository` right away, and internally will only call the dependencies when needed.

Reply:
Yes! 🤦 missed removing that.